src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,171 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<T>(null, value, tail);
+            tail = new Node<T>(head, value, null);
+        }
+        if (size == 1) {
+            tail = new Node<>(head, value, null);
+            head.next = tail;
+        }
+        if (size > 1) {
+            Node<T> main = new Node<>(tail, value, null);
+            tail.next = main;
+            tail = main;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (size == index) {
+            add(value);
+            return;
+        }
+        checkIndexInBounds(index);
+        addBeforeIndex(value, index);
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
+        checkIndexInBounds(index);
+        Node<T> node = getNode(index);
+        if (node != null) {
+            return node.main;
+        }
```

Comment:
```suggestion
        checkIndexInBounds(index);
        Node<T> node = getNode(index);
        return node.value;
```
if index is not valid ```checkIndexInBounds``` would throw an exception
and if it doesnt - there will always be a node by the given index